src/ToggleButton.svelte

```diff
@@ -0,0 +1,72 @@
+<script lang="ts" context="module">
+  import { capitalize } from "@app/utils";
+
+  export interface ToggleButtonOption<T> {
+    count?: number;
+    title?: string;
+    value: T;
+  }
+</script>
+
+<script lang="ts" strictEvents>
+  type T = $$Generic;
+  import { createEventDispatcher } from "svelte";
+
+  export let options: ToggleButtonOption<T>[];
+  export let active: T;
+
+  const dispatch = createEventDispatcher<{ select: T }>();
+
+  function disabled(count?: number) {
+    return count === undefined || count === 0;
+  }
+
+  function onSelect(option: ToggleButtonOption<T>) {
+    if (! disabled(option.count)) {
+      dispatch("select", option.value);
+    }
+  }
+</script>
+
+<style>
+  .wrapper {
+    display: flex;
+    gap: 1rem;
+    user-select: none;
+  }
+  button {
+    border-radius: var(--border-radius-small);
+    color: var(--color-foreground-80);
+    cursor: pointer;
+    font-family: var(--font-family-monospace);
+    font-size: 0.75rem;
+    height: var(--button-tiny-height);
+    padding: 0.25rem 0.5rem;
+    border: none;
+    min-width: 0;
+  }
+  button:hover, button.active {
+    cursor: pointer;
+    color: var(--color-foreground);
+    background-color: var(--color-foreground-background);
+  }
+  button[disabled], button[disabled]:hover {
+    cursor: not-allowed;
+    color: var(--color-foreground-80);
+  }
+</style>
+
+<div class="wrapper">
+  {#each options as option}
+    <button
+      class="state-toggle"
+      on:click={ () => onSelect(option) }
+      disabled={ disabled(option.count) }
+      class:active={ active === option.value }>
```

Comment:
Is there an ESLint rule for spacing in Svelte components?, we don't use spaces in Svelte props in any other component